examples/sim/sr2.py

```diff
@@ -0,0 +1,66 @@
+from migen import *
+
+
+class ShiftRegisterSerialToSerial(Module):
+    """A serial-to-serial shift-right register,
```

Comment:
It's not exactly serial input because the ``values`` input is actually parallel, even if it's read only one bit at a time.
(I also suggest renaming ``values`` to ``i`` or ``input``, in both files)